modules/sql-engine/src/main/java/org/apache/ignite/internal/sql/engine/exec/ExecutionServiceImpl.java

```diff
@@ -367,14 +393,36 @@ private AsyncCursor<InternalSqlRow> executeExecutablePlan(
                 DUMMY_DESCRIPTION,
                 handler,
                 Commons.parametersMap(operationContext.parameters()),
-                TxAttributes.fromTx(tx),
+                TxAttributes.dummy(),
```

Comment:
Because KV operation don't really need execution context. They need only dynamic parameters to build key/value tuple properly, while execution context is part of multistep query execution. I think, it's better to create fake object for this particular case rather than handle potential NPE (or guard it every time with `assert attribute != null` like I did with numerous attributes of operationContext within this patch) everywhere within multistep execution path. Moreover there is already fake fragment description, so one more fake object won't make it worse.

Reply:
> why not just pass null?

well, we can pass null and just suppress the warning, if you insist. But I would prefer to rework this part so this won't be a problem